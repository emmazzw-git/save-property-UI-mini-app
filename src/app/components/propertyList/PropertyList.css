.propertyListContainer {
    display: flex;
    flex-direction: row;
    display: -webkit-flex; /* Safari */
    -webkit-flex-direction: row-reverse;/* Safari 6.1+ */
    width: 400px;
    padding: 20px 0;
    margin: 0 auto;
}
.col {
    text-align: center;
    padding: 0 20px;
}