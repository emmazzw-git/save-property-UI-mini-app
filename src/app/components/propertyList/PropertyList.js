import React, { Component } from 'react'
import PropTypes from 'prop-types'

import PropertyCard from './propertyCard/PropertyCard'
import './PropertyList.css'

export class PropertyList extends Component {
  componentDidMount () {
    this.props.fetchPropertyList()
  }

  render () {
    const { results = [], saved = [] } = this.props
    return (
      <div className='propertyListContainer'>
        <div className='col'>
          <div>Results</div>
          {results.map(result => <PropertyCard propertyDetails={result} key={result.id} />)}
        </div>
        <div className='col'>
          <div>Saved Properties</div>
          {saved.map(s => <PropertyCard propertyDetails={s} key={s.id} />)}
        </div>
      </div>
    )
  }
}

PropertyList.propTypes = {
  results: PropTypes.array,
  saved: PropTypes.array,
  fetchPropertyList: PropTypes.func
}

export default PropertyList
