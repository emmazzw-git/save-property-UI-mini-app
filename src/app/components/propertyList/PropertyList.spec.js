import React from 'react'
import { shallow } from 'enzyme'
import PropertyList from './PropertyList'

describe('PropertyList', () => {
  test('snapshot test with no property details', () => {
    const props = {results: [], saved: [], fetchPropertyList: jest.fn()}
    const component = shallow(<PropertyList {...props} />)
    expect(component).toMatchSnapshot()
  })

  test('snapshot test with property details returned', () => {
    const props = {results: [{id: '1'}], saved: [{id: '2'}], fetchPropertyList: jest.fn()}
    const component = shallow(<PropertyList {...props} />)
    expect(component).toMatchSnapshot()
  })
})
