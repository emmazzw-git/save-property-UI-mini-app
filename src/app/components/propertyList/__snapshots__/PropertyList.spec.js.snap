// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyList snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PropertyList
    fetchPropertyList={[Function]}
    results={Array []}
    saved={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col"
>
          <div>
                    Results
          </div>
</div>,
        <div
          className="col"
>
          <div>
                    Saved Properties
          </div>
</div>,
      ],
      "className": "propertyListContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              Results
</div>,
            Array [],
          ],
          "className": "col",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Results",
            },
            "ref": null,
            "rendered": "Results",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              Saved Properties
</div>,
            Array [],
          ],
          "className": "col",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Saved Properties",
            },
            "ref": null,
            "rendered": "Saved Properties",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col"
>
            <div>
                        Results
            </div>
</div>,
          <div
            className="col"
>
            <div>
                        Saved Properties
            </div>
</div>,
        ],
        "className": "propertyListContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                Results
</div>,
              Array [],
            ],
            "className": "col",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Results",
              },
              "ref": null,
              "rendered": "Results",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                Saved Properties
</div>,
              Array [],
            ],
            "className": "col",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Saved Properties",
              },
              "ref": null,
              "rendered": "Saved Properties",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
